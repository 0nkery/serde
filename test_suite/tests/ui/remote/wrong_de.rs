#[macro_use]
extern crate serde_derive;

mod remote {
    pub struct S(pub u16);
}

#[derive(Deserialize)]
#[serde(remote = "remote::S")]
struct S(u8);

fn main() {}
