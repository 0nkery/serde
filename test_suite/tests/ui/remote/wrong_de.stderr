error[E0308]: mismatched types
  --> $DIR/wrong_de.rs:16:10
   |
16 | #[derive(Deserialize)]
   |          ^^^^^^^^^^^ expected u16, found u8
help: you can cast an `u8` to `u16`, which will zero-extend the source value
   |
16 | #[derive(Deserialize.into())]
   |          ^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
